src/FishyFlip/CborExtensions.cs

```
@@ -180,7 +180,13 @@ internal static class CborExtensions
             {
                 case CBORType.ByteString:
                     var cid = obj.GetByteString();
-                    return Cid.Read(cid);
+                    if (cid[0] != 0)
+                    {
+                        logger?.LogError("ATCid CBOR bytes should start with 0.");
```

Usually I'd throw an exception here, but I saw that the rest of this file also logs and continues on invalid data